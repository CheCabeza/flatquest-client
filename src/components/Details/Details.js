import * as React from 'react';
import {
  View,
  StyleSheet,
  Text,
  Image,
  ScrollView,
  TouchableOpacity,
} from 'react-native';
import {connect} from 'react-redux';
import Icon from 'react-native-vector-icons/FontAwesome';
import Icon2 from 'react-native-vector-icons/MaterialCommunityIcons';
import {PropTypes} from 'prop-types';
import {SafeAreaView} from 'react-native-safe-area-context';
import styles from '../List/styles';

function Details({navigation, flats, user, route}) {
  const {flatId} = route.params;
  return (
    <SafeAreaView style={{paddingBottom: 30}}>
      <ScrollView>
        <View style={styles.page}>
          <Text style={{fontSize: 20}}>Details</Text>
          <View>
            {flats
              .filter(flatList => flatList._id === flatId)
              .map((flatDetails, index) => (
                <View style={styles.box} key={index}>
                  <Image
                    style={styles.img}
                    source={{
                      uri: `${flatDetails.photos[0]}`,
                    }}
                  />
                  <View style={styles.info}>
                    <Text>{flatDetails.address}</Text>
                    <Text>{flatDetails.neighborhood}</Text>
                    <Text>{flatDetails.price} â‚¬</Text>
                    <View style={styles.iconSection}>
                      <Text>{flatDetails.toilet} </Text>
                      <Icon
                        style={styles.icon}
                        name="bath"
                        color="#000"
                        size={25}
                      />
                      <Text>{flatDetails.rooms} </Text>
                      <Icon
                        style={styles.icon}
                        name="bed"
                        color="#000"
                        size={25}
                      />
                      <Text>{flatDetails.m2}m2</Text>
                      <Icon2
                        style={styles.icon}
                        name="tape-measure"
                        color="#000"
                        size={25}
                      />
                      <View
                        style={[styles.icon_container, {top: -85, right: -30}]}>
                        <TouchableOpacity
                          onPress={() =>
                            navigation.navigate('EditFlatReview', {
                              flatId: flatId,
                            })
                          }>
                          <Icon testID="editButton" name="plus" size={25} />
                        </TouchableOpacity>
                      </View>
                    </View>
                  </View>
                  <View style={detailStyles.detailIcons}>
                    <View style={detailStyles.prosIcons}>
                      {flatDetails.pros.map(prosValue => (
                        <Icon
                          key={prosValue}
                          style={styles.icon}
                          name={prosValue}
                          color="#2C7F42"
                          size={40}
                        />
                      ))}
                    </View>
                    <View style={detailStyles.consIcons}>
                      {flatDetails.cons.map(consValue => (
                        <Icon
                          key={consValue}
                          style={styles.icon}
                          name={consValue}
                          color="#CA0303"
                          size={40}
                        />
                      ))}
                    </View>
                  </View>
                  <View style={detailStyles.reviewSection}>
                    {flatDetails.reviews.map(reviewValue => (
                      <Text
                        key={reviewValue[0]}
                        style={detailStyles.reviewText}>
                        {reviewValue}
                      </Text>
                    ))}
                  </View>
                </View>
              ))}
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

Details.propTypes = {
  flats: PropTypes.array.isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps(store) {
  return {
    flats: store.flats,
    user: store.user,
  };
}

export default connect(mapStateToProps)(Details);

const detailStyles = StyleSheet.create({
  detailIcons: {
    display: 'flex',
    flexDirection: 'row',
    margin: 20,
    paddingTop: 10,
    width: 300,
    borderTopWidth: 1,
    borderTopColor: '#C4C4C4',
    borderBottomWidth: 1,
    borderBottomColor: '#C4C4C4',
  },
  prosIcons: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    flex: 1,
    flexWrap: 'wrap',
  },
  consIcons: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    flex: 1,
    flexWrap: 'wrap',
  },
  reviewSection: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  },
  reviewText: {
    width: 300,
    height: 'auto',
    alignSelf: 'center',
    paddingLeft: 15,
    padding: 13,
    margin: 10,
    borderWidth: 2,
    borderRadius: 25,
    borderColor: '#397ED0',
  },
});
