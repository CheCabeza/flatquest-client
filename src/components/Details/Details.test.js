import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import Details from './Details';
import thunk from 'redux-thunk';

describe('Given a Details component', () => {
  const mockStore = configureStore([thunk]);
  let store;
  let navigation;
  let route;

  beforeEach(() => {
    store = mockStore({
      user: {
        email: 'ho@la.com',
        pass: '123123123',
      },
      flats: [
        {
          _id: 123,
          photos: 'a',
          pros: ['glass', 'folder-open'],
          cons: ['money', 'copy'],
          reviews: ['Mierda de test'],
        },
      ],
    });
    route = {
      params: {
        flatId: 123,
      },
    };
    navigation = {
      navigate: jest.fn(),
    };
  });

  test('Should render the lobby of page', () => {
    const lobby = render(
      <Provider store={store}>
        <Details route={route} navigation={navigation} />
      </Provider>,
    );
    expect(lobby).toMatchSnapshot();
  });

  test('Then navigation.navigate is invoked', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <Details route={route} navigation={navigation} />
      </Provider>,
    );
    const button = getByTestId('editButton');
    fireEvent.press(button);
    expect(navigation.navigate).toHaveBeenCalled();
  });
});
