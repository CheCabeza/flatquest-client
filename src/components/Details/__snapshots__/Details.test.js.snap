// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Details component Should render the lobby of page 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "paddingBottom": 30,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          Details
        </Text>
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F5F5F5",
                "borderRadius": 20,
                "display": "flex",
                "elevation": 5,
                "justifyContent": "center",
                "margin": 10,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "a",
                }
              }
              style={
                Object {
                  "borderTopLeftRadius": 20,
                  "borderTopRightRadius": 20,
                  "height": 220,
                  "width": 350,
                }
              }
            />
            <View
              style={
                Object {
                  "margin": 10,
                }
              }
            >
              <Text />
              <Text />
              <Text>
                 €
              </Text>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <Text>
                   
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 25,
                      },
                      Object {
                        "margin": 10,
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <Text>
                   
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 25,
                      },
                      Object {
                        "margin": 10,
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <Text>
                  m2
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 25,
                      },
                      Object {
                        "margin": 10,
                      },
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰭍
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#F5F5F5",
                        "borderRadius": 50,
                        "display": "flex",
                        "height": 45,
                        "justifyContent": "center",
                        "position": "relative",
                        "width": 45,
                      },
                      Object {
                        "right": -30,
                        "top": -85,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="editButton"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderBottomColor": "#C4C4C4",
                  "borderBottomWidth": 1,
                  "borderTopColor": "#C4C4C4",
                  "borderTopWidth": 1,
                  "display": "flex",
                  "flexDirection": "row",
                  "margin": 20,
                  "paddingTop": 10,
                  "width": 300,
                }
              }
            >
              <View
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#2C7F42",
                        "fontSize": 40,
                      },
                      Object {
                        "margin": 10,
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#2C7F42",
                        "fontSize": 40,
                      },
                      Object {
                        "margin": 10,
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#CA0303",
                        "fontSize": 40,
                      },
                      Object {
                        "margin": 10,
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#CA0303",
                        "fontSize": 40,
                      },
                      Object {
                        "margin": 10,
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "borderColor": "#397ED0",
                    "borderRadius": 25,
                    "borderWidth": 2,
                    "height": "auto",
                    "margin": 10,
                    "padding": 13,
                    "paddingLeft": 15,
                    "width": 300,
                  }
                }
              >
                Mierda de test
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
